use-shopping-cart/src/reducers.js

```diff
@@ -48,25 +48,59 @@ export const cartValuesInitialState = {
   totalPrice: 0,
   cartCount: 0
 }
-function Entry(productData, quantity, currency, language) {
+
+function Entry(
+  product,
+  quantity,
+  currency,
+  language,
+  price_metadata,
+  product_metadata
+) {
   const getProductId = () => {
-    if (productData.id) {
-      return productData.id
+    if (product.id) {
+      return product.id
     }
-    if (productData.price_id) {
-      return productData.price_id
+    if (product.price_id) {
+      return product.price_id
     }
-    if (productData.sku_id) {
-      return productData.sku_id
+    if (product.sku_id) {
+      return product.sku_id
     }
-    if (productData.sku) {
-      return productData.sku
+    if (product.sku) {
+      return product.sku
     }
     return uuidv4()
   }
 
+  if (!product.price_data && price_metadata) {
+    product.price_data = {}
+
+    product.price_data = {
+      ...price_metadata
+    }
+  }
+
+  if (product.price_data && price_metadata) {
```

Comment:
Should probably be `else if` here because one should only happen if the other isn't going to.

Reply:
Oh, I see. So here I could add the price_data key to product if it didn't exist, yeah?

Lemme know if I'm misunderstanding how stuff works

Reply:
Your code inside the if-statement on line 84 is fine, you're taking the existing `product.price_data` and merging it with `price_metadata`.

But, if you run through line-by-line from line 76, you're doing the following (assuming no product price_data already existed):

- Line 76: `product.price_data` is falsey and `price_metadata` is truthy
- Line 79: So we shall copy the `price_metadata` over to the product
- Since the two if-statements are separate if the first condition is true, the second one is still evaluated
- Line 84: `product.price_data` is truthy and `price_metadata` is truthy (condition passed, which we don't want)
- Line 85: merging `price_metadata` with `product.price_data` (doesn't need to happen since, originally, there was no `product.price_data`)